plugins/network-elements/vyos/README.md

```diff
@@ -0,0 +1,188 @@
+# Vyos Router Plugin
+
+Vyos (formerly Vyatta) is an open source routing platform built upon Debian Linux.
+It has been modeled to operate in a manner that mirrors more traditional enterprise
+grade routing hardware. It has a lively development community as well as a track
+record of stable production operation stretching back many years. Vyos can
+efficiently operate in production environments either deployed to bare metal or
+as a VM.
+  The Vyos Project website can be found [here](https://vyos.io/ "https://vyos.io/") .
+
+
+## Introduction to the Plugin
+
+The ultimate goal of this plugin is to provide a viable drop in alternative to the
+current VRouter. Vyos should be able to reach almost 100% feature parity with the
+VRouter without having to make any changes to Vyos itself. This has the significant
+advantage of providing end users with the option to use Vyos or the VRouter in
+production as they see fit. For Cloudstack, using Vyos means that there is a
+lively and dedicated community of developers focused solely on the routing
+functionality. This should help to ensure that issues on the router portion of the
+Cloudstack ecosystem are addressed promptly by developers who have years of experience
+contributing to networking projects.
+
+## Current Release
+
+The current version of the Vyos Plugin should be considered a proof of concept only.
+It has some missing features, and design limitations that will need to be
+addressed before it will be ready for review as a candidate for inclusion in cloudstack
+proper.
+
+This first release, (including this document) was developed using the existing Palo Alto Firewall plugin
+as a pattern. The goal was to produce a proof of concept release that attained feature parity with the Palo
+Alto plugin for review and comment by the community. The remaining features of the VRouter will then be
+added to the plugin in an iterative manner.
+
+### Supported Operations
+
+  * List/Add/Delete Vyos service provider
+  * List/Add/Delete Vyos network service offering
+  * List/Add/Delete Vyos network with above service offering
+  * Implement Guest Network
+  * List/Add/Delete Ingress Firewall rule
+  * List/Add/Delete Egress Firewall rule
+  * List/Add/Delete Port Forwarding rule
+  * List/Add/Delete Static Nat rule
+  * Shutdown Guest Network
+  * Basic Unit Tests
+  * Basic Functional Tests
+  * Basic Integration Tests
+
+### Limitations and Design Liabilities
+
+  * **SSH Access to the Vyos Router is not properly secured. It is open to all incoming
```

Comment:
Here's how you can do, maybe similar to how it's done in the ACS Virtual Router: (1) make ssh server run/listen on a non-standard port, such as 3922, (2) configure sshd to allow authentication to root user based on public key see VR's sshd config for reference, (3) have a init/systemd file that can read/copy ssh-public keys from a mounted iso file (VR uses systemvm.iso file, the boot process attaches this iso, and VR patches itself using files in the iso), or a secured location, or do a first-time-setup-thing where you ssh into the VR with known credentials and setup security etc.